configs/components/ruby-2.4.2.rb

```diff
@@ -0,0 +1,300 @@
+component 'ruby-2.4.2' do |pkg, settings, platform|
+  pkg.version '2.4.2'
+  pkg.md5sum 'fe106eed9738c4e03813ab904f8d891c'
+  pkg.url "https://cache.ruby-lang.org/pub/ruby/2.4/ruby-#{pkg.get_version}.tar.gz"
+  pkg.mirror "#{settings[:buildsources_url]}/ruby-#{pkg.get_version}.tar.gz"
+
+  if platform.is_windows?
+    pkg.add_source 'file://resources/files/ruby_242/windows_ruby_gem_wrapper.bat'
+  end
+
+  pkg.replaces 'pe-ruby'
+  pkg.replaces 'pe-ruby-mysql'
+  pkg.replaces 'pe-rubygems'
+  pkg.replaces 'pe-libyaml'
+  pkg.replaces 'pe-libldap'
+  pkg.replaces 'pe-ruby-ldap'
+  pkg.replaces 'pe-rubygem-gem2rpm'
```

Comment:
I'm going to have to think about this more, if it makes sense to keep these in the runtime or if they should move into the runtime component added to the runtime component in puppet-agent. I'm pretty sure as it is these won't persist into the p-a package